internal/db/modelmigration.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// AddModelMigration stores information about an incoming model migration.
+//   - returns an error with code errors.CodeAlreadyExists if
+//     a migration row with the same model UUID already exists.
+func (d *Database) AddModelMigration(ctx context.Context, modelMigration *dbmodel.ModelMigration) (err error) {
+	const op = errors.Op("db.AddModelMigration")
+	if err := d.ready(); err != nil {
+		return errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	db := d.DB.WithContext(ctx)
+
+	if err := db.Create(modelMigration).Error; err != nil {
+		return errors.E(op, dbError(err))
+	}
+	return nil
+}
+
+// GetModelMigration returns model migration information based on the model UUID.
+func (d *Database) GetModelMigration(ctx context.Context, modelMigration *dbmodel.ModelMigration) (err error) {
```

Comment:
I know we usually pass the dbmodel obj, but would love to see us just return a dbmodel.ModelMigration and the func take a uuid. Really hate this style and want us to move away from it honestly.

Reply:
I'd prefer to just follow the pattern, so leaving this as is.